src/hooks/useDebounceEvent/index.ts

```diff
@@ -1,10 +1,26 @@
-import { useMemo, useCallback, useState, useRef, useEffect } from "react";
-import debounce from "lodash/debounce";
-import noop from "lodash/noop";
+import { useMemo, useCallback, useState, useRef, useEffect, ChangeEvent, Dispatch, SetStateAction } from "react";
+import { noop, debounce } from "lodash";
 
-export default function useDebounceEvent({ delay = 0, onChange = noop, initialStateValue = "", trim }) {
-  const [inputValue, setValue] = useState(initialStateValue);
-  const previousValue = useRef(null);
+export type useDebounceResult = {
+  inputValue: string;
+  onEventChanged: (event: ChangeEvent<Partial<HTMLInputElement> | Partial<HTMLTextAreaElement>>) => void;
+  clearValue: () => void;
+  updateValue: Dispatch<SetStateAction<string>>;
+};
+
+export default function useDebounceEvent({
+  delay = 0,
+  onChange = noop,
+  initialStateValue = "",
+  trim
+}: {
+  delay?: number;
+  onChange: (value: string) => void;
```

Comment:
```suggestion
  onChange?: (value: string) => void;
```

Or lose the default value.